doc/source/res/query.md

```diff
@@ -0,0 +1,175 @@
+
+Inspired by ElasticSearch, Milvus provides a Query DSL(Domain Specific Language) consisting of two types of clauses to define queries:
+
+
+**Leaf query clauses**
+
+Leaf query clauses look for a particular value in a particular field. Currently milvus support `term`, `range`, `vector` queries.
+
+  * term: term query matches the entities which corresponding field value are in the specified list. The format is `term: [$value1, $value2, ...]`
+
+
+  * range: range query matches the entities which corresponding field value are in the specified range. The supported range types are:
+
+    - "GT"(greater than)
+    - "GTE"(greater than or equal)
+    - "LT"(less than)
+    - "LTE"(less than or equal).
+
+
+  * vector: vector query only take effect on vector fields, and approximately search nearest vectors. The format should be `{"topk": $topk, "query": $vectors, "params": {...}, "metric_type": $metric}`.
+
+    here, "topk" is the number of approximately nearest top-k entities for each query vector. "query" is a list of query vectors. "params" is search parameters, and "metric_type" indicates which distance computation type to be used.
+
+Note that, the `term` and `range` query act as filter queries for `vector` query, they are pre-filtered.
+
+**Compound query clauses**
+
+Currently, milvus only support boolean query, i.e. `bool`. There are three occurrence types:
```

Comment:
Grammarly, `occurrence` means an incident or event, "There are three types" maybe better.

Reply:
The name `occurrence` is copied from ES.

Reply:
Then we found a bug. < - <

Reply:
I trust google more than ES < - <